sdk/keyvault/azure-security-keyvault-certificates/inc/azure/keyvault/certificates/certificate_client_models.hpp

```diff
@@ -0,0 +1,555 @@
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateKeyUsage AutoRenew;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificateKeyUsage EmailContacts;
+  };
+
+  class CertificatePolicyAction final {
+  private:
+    std::string m_value;
+
+  public:
+    /**
+     * @brief Construct a new Key Curve Name object.
+     *
+     * @param value The string value of the instance.
+     */
+    explicit CertificatePolicyAction(std::string value)
+    {
+      if (value.empty())
+      {
+        throw std::invalid_argument("The value for the curve name can not be empty");
+      }
+      m_value = std::move(value);
+    }
+
+    /**
+     * @brief Construct a default key curve.
+     *
+     */
+    CertificatePolicyAction() = default;
+
+    /**
+     * @brief Enables using the equal operator for key curve.
+     *
+     * @param other A key curve to be compared.
+     */
+    bool operator==(const CertificatePolicyAction& other) const noexcept
+    {
+      return m_value == other.m_value;
+    }
+
+    /**
+     * @brief Get the string value of the key curve.
+     *
+     */
+    std::string const& ToString() const { return m_value; }
+
+    /**
+     * @brief Gets the NIST P-256 elliptic curve, AKA SECG curve SECP256R1.
+     *
+     * @remark For more information, see
+     *  <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction
+        DigitalSignature;
+
+    /**
+     * @brief Gets the SECG SECP256K1 elliptic curve.
+     *
+     * @remark For more information, see
+     * <a href="https://docs.microsoft.com/azure/key-vault/keys/about-keys#curve-types">Curve
+     * types</a>.
+     *
+     */
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction NonRepudiation;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction
+        KeyEncipherment;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction
+        DataEncipherment;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction KeyAgreement;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction KeyCertSign;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction CrlSign;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction EncipherOnly;
+    AZ_SECURITY_KEYVAULT_CERTIFICATES_DLLEXPORT static const CertificatePolicyAction DecipherOnly;
+  };
+
+  struct LifetimeAction
+  {
+    CertificatePolicyAction Action;
+    Azure::Nullable<int32_t> DaysBeforeExpiry;
+    Azure::Nullable<int32_t> LifetimePercentage;
+  };
+
+  struct IssuerProperties
+  {
+    Azure::Nullable<bool> CertTransparency;
+    Azure::Nullable<std::string> Cty;
```

Comment:
Does the name `Cty` come from Azure/Swagger, and does it match other language SDK? No idea what "cty" means :)

Reply:
This struct shouldn't even be exposed, actually.

Reply:
It is in the REST documentation:
https://docs.microsoft.com/en-us/rest/api/keyvault/get-certificate/get-certificate#issuerparameters

It is exposed in .NET as ContentType directly from the `Policy`.  .NET decided to move Cty ,CertTransparency  and issuerName to the policy directly (flat struct).  I decided to follow the REST structure here instead.

Reply:
Should we name it `CertificateTransparency` then?